packages/realm/src/ProgressRealmPromise.ts

```diff
@@ -167,18 +175,31 @@ export class ProgressRealmPromise implements Promise<Realm> {
    */
   progress(callback: ProgressNotificationCallback): this {
     this.listeners.add(callback);
+    if (callback.length === 1) {
+      const estimateCallback = callback as DynamicProgressNotificationCallback;
+      estimateCallback(1.0);
+    } else {
+      const pbsCallback = callback as PartitionBasedSyncProgressNotificationCallback;
+      pbsCallback(0.0, 0.0);
+    }
```

Comment:
I think the manual trigger was to imitate past behavior, it may only be relevant for `estimateCallback` rather than `pbsCallback` since I believe before .progresss would usually have these values at initialization, I didn't look too much into this as this was hitting a previous. Agree with adding a comment here to clear up why.